pkg/scaffold/build_dockerfile.go

```diff
@@ -24,21 +24,46 @@ const DockerfileFile = "Dockerfile"
 
 type Dockerfile struct {
 	input.Input
+
+	// Multistage determines if a multistage Dockerfile template is used.
+	Multistage bool
 }
 
 func (s *Dockerfile) GetInput() (input.Input, error) {
 	if s.Path == "" {
 		s.Path = filepath.Join(BuildDir, DockerfileFile)
 	}
-	s.TemplateBody = dockerfileTmpl
+	if s.Multistage {
+		s.TemplateBody = dockerfileMultiTmpl
+	} else {
+		s.TemplateBody = dockerfileNonMultiTmpl
+	}
+	s.IfExistsAction = input.Error
 	return s.Input, nil
 }
 
-const dockerfileTmpl = `FROM alpine:3.8
+const dockerfileMultiTmpl = `# Binary builder image
+FROM golang:1.10.3 AS builder
```

Comment:
Any reason behind choosing `1.10.3`?

Reply:
No, and this should be updated to `1.10` to get the latest go 1.10 image.